### DISCUSSION TITLE
Theming with python library

### DISCUSSION THREAD

**Discussion by anwar3606** [mathoudebine/turing-smart-screen-python#113] [Themes]
Theming with YAML file is cool but limited. I used their lcd library to create this.
Any idea how can I share the python code with the community? 

![20221217_072453](https://user-images.githubusercontent.com/3619250/208974216-76aed1a0-8f14-4b47-9945-ff58b0715b07.jpg)

![20221218_012416](https://user-images.githubusercontent.com/3619250/208974395-87c78e51-f661-41d5-a1de-e72284157b27.jpg)

**Comment by Vetro05** [comment]
man im interested in this code, telll me when you share it

**Comment by mathoudebine** [comment]
Hi @anwar3606
Very nice work with this library !
I'm not sure it can be easily integrated to this project, but if you create a Github repository with your work I can share it on the Readme of this project under a "Other projects" section maybe?

**Reply by RussNelson** [reply]
I think it could be integrated. What if the YAML went away, to be replaced by a set of calls to a "spec" routine which takes the same parameters that are in the YAML? It could be used to construct the same THEME_DATA data structure, but it would be easier to have default values that don't need to be specified.

**Comment by Cr4zyM1K4** [comment]
Looks great, especially the music view.
May I ask if you have shared the files anywhere?

**Comment by Jeanbonblanc** [comment]
Hello, this crazy layout nowhere to be found?

**Comment by fern4nd0sb** [comment]
oh my... that looks great, I'd love to know how to do a theme like this

**Comment by ooemir23** [comment]
Pls share bro.

**Comment by bitbyte90** [comment]
It really seems most people share the Python coding by just copy/pasting it in the body on the message - within your forum subject post.
So yeah, just share the text, we can all get it that way. 

It is great work btw, you've inspired me to start making some themes too! :D

**Comment by Donwido** [comment]
Hi @anwar3606 You can share the code pls?

**Comment by ElGoleador** [comment]
@anwar3606 may you share with us this code please?, we are waiting since 2022 
thank you

**Comment by Joabutt** [comment]
Hi! I managed to combine the spotify developers api with the UsbPCMonitor 3.5" screen, allowing me to display real time music playing with a progress bar. I did this through the lcd_comm_rev_a library, and it works like a charm. 


# Setup instructions:
1. Clone the base repo 
`$ git clone https://github.com/mathoudebine/turing-smart-screen-python`
2. Install the necessary libraries with `$ pip install -r requirements.txt` and `$ pip install requests spotipy`
3. Copy the code from this [Gist](https://gist.github.com/Joabutt/d4d33c3228975dfdc8f21efdaad5fc37) and put it in the base directory of the cloned repo
4. Go to [Spotify Developer Dashboard](https://developer.spotify.com/dashboard) and create a new app
5. Give the app a name and description, and set the redirect URI to your desired URI, I usually just set it to `http://127.0.0.1:8888/callback` as a default, but it doesn't really matter because it is not being used in this case
6. Then under the `Which API/SDKs are you planning to use` section, tick the Web API
7. Then comply to the Terms of service and create your app
8. Copy the Client Secret, Client ID and the redirect URI then put it into the code
9. Run the code and Enjoy! 

## If you have any issues/questions, feel free to contact me on discord @joabutt or drop me an email at me@joabl.ee

![1](https://github.com/user-attachments/assets/57310f41-8123-4a0f-9d62-518529f7ec4c)
![2](https://github.com/user-attachments/assets/c9c1a729-b054-420a-89aa-4b19bd00813c)
